jwql/website/apps/jwql/urls.py

```diff
@@ -39,18 +39,33 @@
 """
 
 from django.urls import path
+from django.urls import re_path
 
+from . import api_views
 from . import views
 
 app_name = 'jwql'
+
+api_instruments = 'nircam|NIRCam|niriss|NIRISS|nirspec|NIRSpec|miri|MIRI|fgs|FGS'
+
 urlpatterns = [
     path('', views.home, name='home'),
     path('about/', views.about, name='about'),
     path('dashboard/', views.dashboard, name='dashboard'),
-    path('<str:inst>/', views.instrument, name='instrument'),
-    path('<str:inst>/archive/', views.archived_proposals, name='archive'),
-    path('<str:inst>/unlooked/', views.unlooked_images, name='unlooked'),
-    path('<str:inst>/<str:file_root>/', views.view_image, name='view_image'),
-    path('<str:inst>/<str:file>/hdr/', views.view_header, name='view_header'),
-    path('<str:inst>/archive/<str:proposal>', views.archive_thumbnails, name='archive_thumb'),
+    path('jwql/<str:inst>/', views.instrument, name='instrument'),
+    path('jwql/<str:inst>/archive/', views.archived_proposals, name='archive'),
+    path('jwql/<str:inst>/unlooked/', views.unlooked_images, name='unlooked'),
+    path('jwql/<str:inst>/<str:file_root>/', views.view_image, name='view_image'),
+    path('jwql/<str:inst>/<str:file>/hdr/', views.view_header, name='view_header'),
+    path('jwql/<str:inst>/archive/<str:proposal>', views.archive_thumbnails, name='archive_thumb'),
+    path('api/proposals/', api_views.all_proposals, name='all_proposals'),
+    path('api/<str:inst>/proposals/', api_views.instrument_proposals, name='instrument_proposals'),
+    re_path(r'^api/(?P<inst>({}))/preview_images/$'.format(api_instruments), api_views.preview_images_by_instrument, name='preview_images_by_instrument'),
```

Comment:
I had to look up the magic that you used to do this, but it is SO cool!!! 
I'm thinking it might be worthwhile to implement the same kind of regex paths for the instrument variables in the non-API URLs, to make them more secure (prevent cross-site scripting)? (That could also be a separate PR if you think that's more appropriate.)

Reply:
@laurenmarietta Agreed!  Let's tackle that in our newly-conceived "make the URLs better" future PR.